registry/zookeeper/registry.go

```diff
@@ -255,3 +264,40 @@ func (r *zkRegistry) getListener(conf *common.URL) (*RegistryConfigurationListen
 
 	return zkListener, nil
 }
+
+func (r *zkRegistry) getCloseListener(conf *common.URL) (*RegistryConfigurationListener, error) {
+
+	var zkListener *RegistryConfigurationListener
+	dataListener := r.dataListener
+	dataListener.mutex.Lock()
+	defer dataListener.mutex.Unlock()
```

Comment:
The datalistener's lock is so long.May be change to rwlock and make the lock time shorter.

Reply:
check it again